configure-ironic.sh

```diff
@@ -2,6 +2,10 @@
 
 . /bin/ironic-common.sh
 
+USE_HTTP_BASIC=${USE_HTTP_BASIC:-false}
+IRONIC_HTTP_BASIC_USERNAME=${IRONIC_HTTP_BASIC_USERNAME:-"change_me"}
+IRONIC_HTTP_BASIC_PASSWORD=${IRONIC_HTTP_BASIC_PASSWORD:-"change_me"}
```

Comment:
IME having these kinds of defaults eventually results in vulnerabilities where a non-default value isn't passed and the default are used by mistake - was it discussed on the upstream PR at all?

Not blocking here, but perhaps we could have a default (empty string?) which causes the container to fail on startup, instead of silently using a predictable default?

Reply:
The only thing I remember was the idea to have a variable to handle the type of auth, but nothing related to the user/password.., I only used the idea of the mariadb password https://github.com/metal3-io/ironic-image/blob/master/configure-ironic.sh#L10, but I can update this to be default to empty string

Reply:
I'd prefer we explicitly fail if IRONIC_HTTP_BASIC_USERNAME is non-empty and IRONIC_HTTP_BASIC_PASSWORD is empty

Reply:
Yeah we should probably remove that default for mariadb too IMO, I'm guessing it was added to make initial prototyping easier, but it's a bad idea for production containers.